File: docs/snippets/IO/spreadsheet-io/investigation-xlsx-file.ts
// docs:begin
import { ArcInvestigation, XlsxController } from "@nfdi4plants/arctrl";

const investigation = new ArcInvestigation(
  "Investigation-XLSX",
  "Spreadsheet IO example",
);

const xlsxPath =
  "docs/generated/snippets/IO/spreadsheet-io/isa.investigation.xlsx";

await XlsxController.Investigation.toXlsxFileAsync(xlsxPath, investigation);

const investigationAgain =
  await XlsxController.Investigation.fromXlsxFileAsync(xlsxPath);
// docs:end

// docs:assert
if (investigationAgain.Identifier !== "Investigation-XLSX") {
  throw new Error(
    `Expected identifier to survive xlsx file roundtrip, got ${investigationAgain.Identifier}`,
  );
}
// docs:endassert

File: docs/snippets/IO/spreadsheet-io/investigation-xlsx-file.py
# docs:begin
from pathlib import Path

from arctrl import ArcInvestigation, XlsxController

investigation = ArcInvestigation(
    "Investigation-XLSX",
    "Spreadsheet IO example",
)

xlsx_path = Path(__file__).with_name("isa.investigation.xlsx")

XlsxController.Investigation().to_xlsx_file(str(xlsx_path), investigation)

investigation_again = XlsxController.Investigation().from_xlsx_file(str(xlsx_path))
# docs:end

# docs:assert
if investigation_again.Identifier != "Investigation-XLSX":
    raise Exception(
        "Expected identifier to survive xlsx file roundtrip, "
        f"got {investigation_again.Identifier}"
    )
# docs:endassert
